INTITA/Modules/Profile/ProfileViewController.swift

```diff
@@ -56,6 +84,119 @@ class ProfileViewController: UIViewController, Storyboarded {
+    @objc func handleGesture(gesture: UIPanGestureRecognizer) {
+        switch gesture.state {
+        case .began:
+            currentTableViewContentYOffset = tableView.contentOffset.y
+            initializeAnimator()
+            animator.startAnimation()
+            animator.pauseAnimation()
+        case .changed:
+            switch headerState {
+            case .normal:
+                animator.fractionComplete = -gesture.translation(in: view).y / 100
+            case .decreased:
+                if tableView.contentOffset.y <= 100 {
+                    animator.fractionComplete = gesture.translation(in: view).y / 100
+                }
+            }
+            if -gesture.translation(in: view).y + currentTableViewContentYOffset > 0, -gesture.translation(in: view).y + currentTableViewContentYOffset < tableView.contentSize.height - tableView.bounds.size.height {
+                tableView.contentOffset.y = -gesture.translation(in: view).y + currentTableViewContentYOffset
+            }
+            
+        case .ended:
+            if animator.fractionComplete >= 0.5 {
+                animator.continueAnimation(withTimingParameters: nil, durationFactor: 0)
+                switch headerState {
+                case .normal:
+                    if tableView.contentOffset.y < 100 + currentTableViewContentYOffset {
+                        if tableView.contentOffset.y + 100 + currentTableViewContentYOffset < tableView.contentSize.height - tableView.bounds.size.height {
+                            tableView.setContentOffset(CGPoint(x: 0, y: tableView.contentOffset.y - currentTableViewContentYOffset + 100), animated: true)
+                        } else {
+                            tableView.scrollToRow(at: IndexPath(row: 4, section: 0), at: .bottom, animated: true)
+                        }
+                    }
+                case .decreased:
+                    tableView.scrollToRow(at: IndexPath(row: 0, section: 0), at: .top, animated: true)
+                }
+                headerState.toggle()
+            } else {
+                animator.isReversed = true
+                animator.startAnimation()
+                if headerState == .normal {
+                    tableView.setContentOffset(CGPoint(x: 0, y: currentTableViewContentYOffset), animated: true)
+                } else {
+                    tableView.setContentOffset(CGPoint(x: 0, y: 100), animated: true)
+                }
+            }
+            animator.addCompletion { [unowned self]_ in
+                var opacity: Float = 0
+                switch headerState {
+                case .decreased:
+                    opacity = 0
+                case .normal:
+                    opacity = 1
+                }
+                headerContentView.layer.sublayers?[3].opacity = opacity
+                headerContentView.layer.sublayers?[4].opacity = opacity
+                headerContentView.layer.sublayers?[5].opacity = opacity
+            }
```

Comment:
animator.addCompletion { [weak self] _ in
guard let self ....
Please take a look.